bin/parse_tomte_qc.py

```diff
@@ -220,19 +245,143 @@ def calculate_slope(coverage_values: List[float]) -> float:
+
+    Example:
+        Example data file content:
+        ```
+        Sample\ttotal_reads\tavg_input_read_length\tuniquely_mapped\t....
+        Sample-01\t100000\t80\t10000\t...
+        Sample-02\t200000\t90\t20000\t...
+        ```
+
+    """
+    columns_needed: Dict[str, Union[type, str]] = {
+        "total_reads": int,
+        "num_noncanonical_splices": int,
+        "num_splices": int,
+        "multimapped_percent": float,
+        "unmapped_mismatches_percent": float,
+        "mapped_percent": float,
+    }
+
+    new_column_names: Dict[str, str] = {
+        "total_reads": "tot_reads",
+        "num_noncanonical_splices": "non_canon_splice",
+        "num_splices": "canon_splice",
+        "multimapped_percent": "multimap_pct",
+        "unmapped_mismatches_percent": "mismatch_pct",
+        "mapped_percent": "mapped_pct",
+    }
+
+    with open(multiqc_star) as file:
+        lines = file.readlines()
+        header = lines[0].strip().split("\t")
+        data = lines[1:]
+
+    sliced_data = {}
+    indices = {}
+    for col in columns_needed.keys():
+        if col not in header:
+            raise ValueError(f"Column {col} not found in {multiqc_star}")
+        indices[col] = header.index(col)
+
+    for line in data:
+        col = line.strip().split("\t")
+        key = col[0]
+        values = {}
+        for col_name, col_type in columns_needed.items():
+            value = col[indices[col_name]]
+            try:
+                if col_type is int and "." in value:
+                    values[new_column_names[col_name]] = int(float(value))
+                else:
+                    values[new_column_names[col_name]] = col_type(value)
+            except ValueError:
+                values[new_column_names[col_name]] = float(value)
+        sliced_data[key] = values
+
+        # Calculate splice_ratio
+        sliced_data[key]["splice_ratio"] = int(
+            round(
+                (
+                    sliced_data[key]["canon_splice"]
+                    / sliced_data[key]["non_canon_splice"]
+                ),
+                0,
+            )
+        )
+
+    return sliced_data
+
+
+def process_hb_estimate_data(
+    hb_estimate: str,
+) -> Dict[str, Dict[str, Union[int, float]]]:
+    """
+    Process merged HB estimate data.
+    """
+    # {"nbr_genes": 63086, "total_count": 14404757, "average_count": 228.335240782424, "target_genes_count": 1932, "target_genes_frac": 0.00013412235971769603, "per_target_gene_count": {"ENSG00000244734": 1679, "ENSG00000223609": 1, "ENSG00000213934": 0, "ENSG00000196565": 0, "ENSG00000213931": 0, "ENSG00000130656": 2, "ENSG00000188536": 189, "ENSG00000206172": 61}}
+    # data = {}
+    # with open(hb_estimate) as file:
+    #     for line in file:
+    #         sample_id, json_data = line.strip().split("\t", 1)
+    #         data[sample_id] = json.loads(json_data)
+    # return data
+    data = {}
+    hb_sample_id = hb_estimate.split("_perc_mapping.json")[0]
```

Comment:
I think using `replace` would be cleaner here

```suggestion
    hb_sample_id = hb_estimate.replace("_perc_mapping.json", "")
```